### DISCUSSION TITLE
How to isolate custom admin api interface.

### DISCUSSION THREAD

**Discussion by pzhx521** [Kong/kong#10187] [Help]
hi.

I have developed a custom plugin in which an extension to the admin api is implemented.
I currently have such a requirement that I want to isolate the port between the custom API interface and the **Kong Admin API**.
The desired effect is as follows:
  `http://127.0.0.1:8001/services` This is a **Kong Admin API**, using the default port.
 ` http://127.0.0.1:8080/my-plugin-api `This is a custom api, using custom configurable other ports.
But I don't know how to implement this idea. Hope to give some guidance.
If the above requirements cannot be met, how can I strengthen the protection of the admin api.
Thanks!

**Comment by locao** [comment]
Hi @pzhx521!

You can restrict the Admin API only to the local network interface and use Kong to proxy to the Admin API itself. Then you can add a security-control mechanism, such as [IP restriction](https://docs.konghq.com/hub/kong-inc/ip-restriction/) or [basic auth](https://docs.konghq.com/hub/kong-inc/basic-auth/).
